libs/wingc/src/type_check.rs

```diff
@@ -1073,6 +1073,17 @@ impl<'a> TypeChecker<'a> {
 				} else if op.numerical_args() {
 					self.validate_type(ltype, self.types.number(), left);
 					self.validate_type(rtype, self.types.number(), right);
+				} else if matches!(op, crate::ast::BinaryOperator::UnwrapOr) {
+					// Left argument must be an optional type
+					if !ltype.is_option() {
+						self.expr_error(left, "Left argument of ?? must be an optional type".to_string());
```

Comment:
I don't think this error needs to mention the visible context. Maybe something along these lines:

```suggestion
						self.expr_error(left, format!("Expected optional type, found \"{}\"", ltype));
```